optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterGenerator.java

```diff
@@ -432,6 +438,50 @@ private Integer createTenant(int spotListSize, int employeeListSize, int lengthI
         return employeeList;
     }
 
+    private List<EmployeeAvailability> createEmployeeAvailabilityList(int tenantId, TenantConfiguration config, List<Employee> employeeList, LocalDate fromDate, LocalDate toDate) {
+        List<EmployeeAvailability> out = new ArrayList<>();
+        List<LocalDate> datesBetween = new ArrayList<>();
+        for (LocalDate currDate = fromDate; currDate.isBefore(toDate); currDate = currDate.plusDays(1)) {
+            datesBetween.add(currDate);
+        }
+        for (LocalDate date : datesBetween) {
+            List<Employee> employeesListCopy = new ArrayList<>(employeeList);
+            List<Employee> unavailableEmployees = new ArrayList<>(extractRandomSubList(employeesListCopy, 0.3));
+            employeesListCopy.removeAll(unavailableEmployees);
+            List<Employee> undesiredEmployees = new ArrayList<>(extractRandomSubList(employeesListCopy, 0.3));
+            employeesListCopy.removeAll(undesiredEmployees);
+            List<Employee> desiredEmployees = new ArrayList<>(extractRandomSubList(employeesListCopy, 0.3));
+            employeesListCopy.removeAll(desiredEmployees);
+
+            unavailableEmployees.forEach((e) -> {
```

Comment:
For consistency, we use a simple old-fashioned "for ( Foo f : foos)" over "foo.forEach()", unless there's a good reason to use forEach() (such as the reuse of a method).